### ISSUE TITLE
[web] Flutter test suite in the web needs love.

### ISSUE THREAD

**Issue by ditman** [flutter/flutter#143124]
While investigating [this revert](https://github.com/flutter/engine/pull/50404), a couple of concerning things were discovered wrt the Flutter test suite in the web.

1. The revert was caused because some tests failed when the `scene` of the HTML renderer started respecting the "overridden" value of `devicePixelRatio` to report its "logical size". 
    * Bad behavior that **fixed golden tests** and allowed `hostElement` to reland was introduced [here](https://github.com/flutter/engine/pull/38477).
    * This used to work before because we were using the [logical size directly from the `window` JS object](https://github.com/flutter/engine/blame/234ab4c1e9c48c20828fba58c2a0a52d169779ed/lib/web_ui/lib/src/engine/html/scene.dart#L51-L52), which doesn't care about overrides.

2. The [devicePixelRatio overridden by flutter tests](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/web/bootstrap.dart#L252-L253) (3.0) does NOT match the [devicePixelRatio set by the browser when it starts](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/web/chrome.dart#L222). 
    * In fact, the browser is starting always with DPR=1. It should pass `'--force-device-scale-factor=X'` for other DPRs.

3. ~~By hijacking the `_matchers_web.dart` file, it seems that the `isCanvasKit` boolean used there to determine how to match a golden never returns `true` (this might be a regression)~~ Fixed by: https://github.com/flutter/flutter/pull/143128

----

TL;DR: The `flutter test` runner needs some love in the `flutter/flutter` repository.
